dktest.go

```diff
@@ -145,10 +145,7 @@ func waitContainerReady(ctx context.Context, lgr logger, c ContainerInfo,
 	for {
 		select {
 		case <-ticker.C:
-			readyCtx, canceledFunc := context.WithTimeout(ctx, readyTimeout)
-			defer canceledFunc()
-
-			if readyFunc(readyCtx, c) {
+			if runReadyFunc(ctx, c, readyFunc, readyTimeout) {
```

Comment:
Let's use an anonymous function instead.
That way we have the flexibility to can use any local variable and don't need to worry about passing local variables as arguments to another function.